src/ostorlab/serve_app/oxo.py

```diff
@@ -177,8 +177,80 @@ def mutate(
             return DeleteScanMutation(result=True)
 
 
+class PublishAgentGroupMutation(graphene.Mutation):
+    """Create agent group."""
+
+    class Arguments:
+        agent_group = types.AgentGroupCreateInputType(required=True)
+
+    agent_group = graphene.Field(types.AgentGroupType)
+
+    @staticmethod
+    def mutate(
+        root,
+        info: graphql_base.ResolveInfo,
+        agent_group: types.AgentGroupCreateInputType,
+    ) -> "PublishAgentGroupMutation":
+        """Create agent group.
+
+        Args:
+            info (graphql_base.ResolveInfo): GraphQL resolve info.
+            agent_group (types.AgentGroupCreateInputType): Agent group to create.
+
+        Returns:
+            PublishAgentGroupMutation: Publish agent group mutation.
+        """
+
+        group = models.AgentGroup.create(
+            name=agent_group.name,
+            description=agent_group.description,
+            agents=agent_group.agents,
+        )
+        return PublishAgentGroupMutation(agent_group=group)
+
+
+class DeleteAgentGroupMutation(graphene.Mutation):
+    """Delete agent group mutation."""
+
+    class Arguments:
+        agent_group_id = graphene.Int(required=True)
+
+    result = graphene.Boolean()
+
+    @staticmethod
+    def mutate(
+        root,
+        info: graphql_base.ResolveInfo,
+        agent_group_id: int,
+    ) -> "DeleteAgentGroupMutation":
+        """Delete agent group mutation.
+
+        Args:
+            info (graphql_base.ResolveInfo): GraphQL resolve info.
+            agent_group_id (int): Agent group id.
+
+        Returns:
+            DeleteAgentGroupMutation: Delete agent group mutation.
+        """
+        with models.Database() as session:
+            agent_group_query = session.query(models.AgentGroup).filter_by(
+                id=agent_group_id
+            )
+            if agent_group_query.count() == 0:
+                raise graphql.GraphQLError("AgentGroup not found.")
+            agent_group_query.delete()
+            session.commit()
```

Comment:
Do we need to delete all the agents related to this agent group?
I think we should've added `cascade delete` on the models..

Reply:
why ? agents and agent group have many2many relation, so an agent used in this agent group might be used in others, so no need to delete.